src/Codeception/Subscriber/Console.php

```diff
@@ -452,115 +462,81 @@ public function printScenarioTrace(ScenarioDriven $failedTest)
-
-        if ($feature) {
-            $this->message = $this
-                ->message($inProgress ? $feature : ucfirst($feature))
-                ->apply(function ($str) { return str_replace('with data set', "|", $str); } )
-                ->cut($inProgress ? $this->columns[0] + $this->columns[1] - 18 - strlen($filename) : $this->columns[0] - 5 - strlen($filename))
-                ->style('focus')
-                ->prepend($inProgress ? 'Trying to ' : '')
-                ->append(" ($filename)");
-            return $this->message;
+        $filename = basename(Descriptor::getTestFileName($test));
+        if ($filename) {
+            $atMessage = $atMessage
+                ->append($this->options['colors'] ? '' : 'at ')
+                ->append($filename);
         }
-        
-        $this->message = $this
-            ->message("<focus>%s</focus> ")
-            ->prepend($inProgress ? 'Running ' : '')
-            ->with($filename);
-        return $this->message;
-    }
 
-    private function cutNamespace($className)
-    {
-        if (!$this->namespace) {
-            return $className;
-        }
-        if (strpos($className, $this->namespace) === 0) {
-            return substr($className, strlen($this->namespace)+1);
-        }
-        return $className;
-    }
+        $stripDataSet = function ($str) {
+            return str_replace('with data set', "|", $str);
+        };
 
-    protected function writeCurrentTest(\PHPUnit_Framework_SelfDescribing $test)
-    {
-        if (!$this->isDetailed($test) && $this->output->isInteractive()) {
+        if (!$test instanceof Descriptive) {
+            $title = $this->message(str_replace('::', ':', $test->toString()))->apply($stripDataSet);
+            $atMessage->cut($this->width - 4 - strlen($title))->style('info');
             $this
-                ->getTestMessage($test, true)
-                ->append('... ')
+                ->message($title)
+                ->append($atMessage)
+                ->prepend($prefix)
                 ->write();
             return;
         }
-        $this->getTestMessage($test)->write();
+
+        $feature = $test->getName(true);
+        if ($test instanceof TestInterface and $test->getMetadata()->getFeature()) {
+            $feature = $test->getMetadata()->getFeature();
+        }
+        $title = $this->message(ucfirst($feature))->apply($stripDataSet);
+        $atMessage->cut($this->width - 4 - strlen($title))->style('info');
+
+        $this->message($title)
+            ->prepend($prefix)
+            ->append($atMessage)
+            ->write();
     }
 
-    protected function writeFinishedTest(\PHPUnit_Framework_Test $test)
+    protected function writelnFinishedTest(TestEvent $e, Message $result)
     {
+        $test = $e->getTest();
         if ($this->isDetailed($test)) {
             return;
         }
 
+        if ($this->output->isInteractive()) {
+            $this->output->write("\x0D");
```

Comment:
Let me explain. In interactive mode we can control the caret on a line. For a test we can print currently started test:

```
- CurrentTest
```
and when test finishes we replace that line with result + test name + time:

```
✓CurrentTest (0.1s) 
```
so we use \r character to print that line on top of previous one